mlir/lib/Conversion/SCFToOpenMP/SCFToOpenMP.cpp

```diff
@@ -398,11 +400,39 @@ struct ParallelOpLowering : public OpRewritePattern<scf::ParallelOp> {
     // Replace the reduction operations contained in this loop. Must be done
     // here rather than in a separate pattern to have access to the list of
     // reduction variables.
+    unsigned int reductionIndex = 0;
     for (auto [x, y] :
          llvm::zip_equal(reductionVariables, reduce.getOperands())) {
       OpBuilder::InsertionGuard guard(rewriter);
       rewriter.setInsertionPoint(reduce);
-      rewriter.create<omp::ReductionOp>(reduce.getLoc(), y, x);
+      Region &redRegion =
+          ompReductionDecls[reductionIndex].getReductionRegion();
+      assert(redRegion.hasOneBlock() &&
+             "expect reduction region to have one block");
```

Comment:
Please could you add a comment explaining why a reduction region must have only one block, or adding a TODO for multiple blocks.